libraries/app/database_api.cpp

```diff
@@ -790,14 +790,19 @@ vector<optional<account_object>> database_api_impl::get_accounts(const vector<st
    std::transform(account_names_or_ids.begin(), account_names_or_ids.end(), std::back_inserter(result),
                   [this](std::string id_or_name) -> optional<account_object> {
 
-      const account_object* account = get_account_from_string(id_or_name);
-      account_id_type id = account->id;
-      if(auto o = _db.find(id))
-      {
-         subscribe_to_item( id );
-         return *o;
+      try {
+         const account_object *account = get_account_from_string(id_or_name);
```

Comment:
Check whether its `nullptr`.

Reply:
OK, My bad. The issue is `get_account_from_string()` has a FC_ASSERT inside. I think it's better to add another function `find_account_from_string` or so which does NOT throw, or add a parameter to `get_account_from_string()` indicates whether it should throw when the account is not found. Using try-catch on the whole code block is not good practice.

Reply:
By the way, if `get_account_from_string` returned an object already, it does not make sense to call `find` again.

Reply:
thank you for the comments, please check new code when you can at https://github.com/bitshares/bitshares-core/pull/1765/commits/e300264fa8c51c31a145c24d44ef213613daa4ea